github_api.py

```diff
@@ -0,0 +1,81 @@
+import json
+import re
+
+import requests
+from flask import abort
+
+import config
+import routes.overview_page
+
+
+pattern = r"(?:https?://)?(?:www\.)?github\.com/([\w-]+)/([\w-]+)/?"
+
+
+def get_github_urls(links):
+    github_urls = set()
+    for category in links["categories"]:
+        for repo in category["category_repos"]:
+            for link in repo["links"]:
+                if re.match(pattern, link["link_value"]):
+                    github_urls.add(link["link_value"])
+
+    return sorted(list(github_urls), key=lambda x: x.split("/")[-1])
+
+
+def get_open_pull_request():
+    # Get list of GitHub repos from Overview page
+    services_links = routes.overview_page.get_services_links()
+    github_urls = get_github_urls(services_links)
+
+    pull_requests = {}
+    gh_token = config.GITHUB_TOKEN
+
+    # Download open pull requests
+    for url in github_urls:
+        match = re.search(pattern, url)
+        if match:
+            owner = match.group(1)
+            repo_name = match.group(2)
+        
+        url = f"https://api.github.com/repos/{owner}/{repo_name}/pulls"
+        params = {"state": "open"}
+
+        headers = {
+            "Accept": "application/vnd.github.v3+json",
+            "Authorization": f"Bearer {gh_token}",
+        }
+
+        try:
+            response = requests.get(url, params=params, headers=headers)
+
+            if response.status_code == 200:
+                json_data = response.json()
+                if not json_data:
+                    pull_requests[repo_name] = []
+                else:
+                    open_pr_list = []
+                    for pr in json_data:
+                        open_pr_list.append(
+                            {
+                                "number": pr["number"],
+                                "draft": pr["draft"],
+                                "title": pr["title"],
+                                "created_at": pr["created_at"],
+                                "user_login": pr["user"]["login"],
+                                "html_url": pr["html_url"],
+                            }
+                        )
+
+                    pull_requests[repo_name] = open_pr_list
+
+            elif response.status_code == 401:
+                abort(401, "401 Unauthorized - check the GitHub token.")
+
+            response.raise_for_status()
```

Comment:
Instead `raise_for_status` and `except` you might use just `else` and `abort` in it so it won't require try and another level of indentation.

Reply:
But that would not catch issues with requests like timeouts or refused connections.